### DISCUSSION TITLE
Security of tokens

### DISCUSSION THREAD

**Discussion by imdevarsh** [community/community#11073] [Copilot Conversations]
I store very very very secret tokens in my `.env` file for one of my projects (it is also in my gitignore), but once I accidentally had copilot on and started to use it. Would Copilot send the `.env` file to copilot servers?

**Comment by KTibow** [comment]
AFAIK Copilot won't send data about a file unless you do something to trigger a completion in that file, or open another file after that file and run a completion there.

**Reply by imdevarsh** [reply]
Usually I have too many tabs open for files in VSCode, but I thought it sent the full project files to Copilot because it understands the codebase?

**Reply by KTibow** [reply]
No. Codex is a model that takes some code, and predicts what comes next. Check what it's sending to Codex under Terminal > Output > GitHub Copilot.
